x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form_objective_section.tsx

```diff
@@ -18,33 +17,62 @@ import {
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import React, { useEffect } from 'react';
 import { Controller, useFormContext } from 'react-hook-form';
-import type { CreateSLOInput } from '@kbn/slo-schema';
-
-import { SloEditFormObjectiveSectionTimeslices } from './slo_edit_form_objective_section_timeslices';
 import {
   BUDGETING_METHOD_OPTIONS,
   CALENDARALIGNED_TIMEWINDOW_OPTIONS,
   ROLLING_TIMEWINDOW_OPTIONS,
   TIMEWINDOW_TYPE_OPTIONS,
 } from '../constants';
+import { CreateSLOForm } from '../types';
 import { maxWidth } from './slo_edit_form';
+import { SloEditFormObjectiveSectionTimeslices } from './slo_edit_form_objective_section_timeslices';
 
 export function SloEditFormObjectiveSection() {
-  const { control, watch, getFieldState, resetField } = useFormContext<CreateSLOInput>();
+  const {
+    control,
+    watch,
+    getFieldState,
+    setValue,
+    formState: { defaultValues },
+  } = useFormContext<CreateSLOForm>();
   const budgetingSelect = useGeneratedHtmlId({ prefix: 'budgetingSelect' });
   const timeWindowTypeSelect = useGeneratedHtmlId({ prefix: 'timeWindowTypeSelect' });
   const timeWindowSelect = useGeneratedHtmlId({ prefix: 'timeWindowSelect' });
   const timeWindowType = watch('timeWindow.type');
 
+  /**
+   * Two flow to handle: Create and Edit
+   * On create: the default value is set to rolling & 30d (useForm)
+   * When we change the window type (from rolling to calendar for example), we want to select a default duration (picking item 1 in the options)
+   * If we don't, the select will show the option as selected, but the value is still the one from the previous window type, until the user manually changes the value
+   *
+   * On edit: the default value is set with the slo value
+   * When we change the window type, we want to change the selected value as we do in the create flow, but we also want to fallback on the initial default value
+   *
+   */
   useEffect(() => {
-    resetField('timeWindow.duration', {
-      defaultValue:
-        timeWindowType === 'calendarAligned'
-          ? CALENDARALIGNED_TIMEWINDOW_OPTIONS[1].value
-          : ROLLING_TIMEWINDOW_OPTIONS[1].value,
-    });
-  }, [timeWindowType, resetField]);
+    if (timeWindowType === 'calendarAligned') {
+      const exists = CALENDARALIGNED_TIMEWINDOW_OPTIONS.map((opt) => opt.value).includes(
+        defaultValues?.timeWindow?.duration ?? ''
+      );
+      setValue(
+        'timeWindow.duration',
+        // @ts-ignore
+        exists ? defaultValues?.timeWindow?.duration : CALENDARALIGNED_TIMEWINDOW_OPTIONS[1].value
+      );
+    } else {
+      const exists = ROLLING_TIMEWINDOW_OPTIONS.map((opt) => opt.value).includes(
+        defaultValues?.timeWindow?.duration ?? ''
+      );
+      setValue(
+        'timeWindow.duration',
+        // @ts-ignore
+        exists ? defaultValues?.timeWindow?.duration : ROLLING_TIMEWINDOW_OPTIONS[1].value
+      );
+    }
+  }, [timeWindowType, setValue, defaultValues]);
```

Comment:
💬 I feel like we can do better here, but I need a break for this. if you want to try to improve it, I would suggest logging `<pre>{JSON.stringify(watch('timeWindow'), null, 2))}</pre>` above the displayed fields, and then try a few scenarios:
1. Create a new SLO and switch from rolling to calendar aligned (assert duration is reflected in state), then change duration to the other one, then change calendar to rolling
2. Update an existing SLO (try calendar aligned and rolling, with 7d/weekly), see if the state is correctly set initially